operator/pkg/charts/charts.go

```diff
@@ -77,51 +85,88 @@ func mergeHelmConfigs(ctx context.Context, meta *ectypes.ReleaseMetadata, in *cl
-		if in.Spec.HighAvailability {
-			seaweedfsConfig, ok := meta.BuiltinConfigs["seaweedfs"]
-			if ok {
-				combinedConfigs.Charts = append(combinedConfigs.Charts, seaweedfsConfig.Charts...)
-				combinedConfigs.Repositories = append(combinedConfigs.Repositories, seaweedfsConfig.Repositories...)
-			}
-
-			migrationStatus := k8sutil.CheckConditionStatus(in.Status, registry.RegistryMigrationStatusConditionType)
-			if migrationStatus == metav1.ConditionTrue {
-				registryConfig, ok := meta.BuiltinConfigs["registry-ha"]
-				if ok {
-					combinedConfigs.Charts = append(combinedConfigs.Charts, registryConfig.Charts...)
-					combinedConfigs.Repositories = append(combinedConfigs.Repositories, registryConfig.Repositories...)
-				}
-			} else {
-				registryConfig, ok := meta.BuiltinConfigs["registry"]
-				if ok {
-					combinedConfigs.Charts = append(combinedConfigs.Charts, registryConfig.Charts...)
-					combinedConfigs.Repositories = append(combinedConfigs.Repositories, registryConfig.Repositories...)
-				}
-			}
-		} else {
-			registryConfig, ok := meta.BuiltinConfigs["registry"]
-			if ok {
-				combinedConfigs.Charts = append(combinedConfigs.Charts, registryConfig.Charts...)
-				combinedConfigs.Repositories = append(combinedConfigs.Repositories, registryConfig.Repositories...)
-			}
+	//set the cluster ID for the operator chart to use
+	clusterUUID, err := uuid.Parse(in.Spec.ClusterID)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse cluster ID: %w", err)
+	}
+	metrics.SetClusterID(clusterUUID)
+
+	// determine the images to use for the operator chart
+	ecOperatorImage := ""
+	ecUtilsImage := ""
+	for _, image := range images {
+		if strings.Contains(image, "/embedded-cluster-operator-image:") {
+			ecOperatorImage = image
 		}
+		if strings.Contains(image, "/ec-utils:") {
+			ecUtilsImage = image
+		}
+	}
+	if ecOperatorImage == "" {
+		return nil, fmt.Errorf("no embedded-cluster-operator-image found in images")
+	}
+	if ecUtilsImage == "" {
+		return nil, fmt.Errorf("no ec-utils found in images")
 	}
 
-	if in != nil && in.Spec.LicenseInfo != nil && in.Spec.LicenseInfo.IsDisasterRecoverySupported {
-		config, ok := meta.BuiltinConfigs["velero"]
-		if ok {
-			combinedConfigs.Charts = append(combinedConfigs.Charts, config.Charts...)
-			combinedConfigs.Repositories = append(combinedConfigs.Repositories, config.Repositories...)
+	embeddedclusteroperator.Metadata.Images = map[string]release.AddonImage{
+		"embedded-cluster-operator": {
+			Repo: strings.Split(ecOperatorImage, ":")[0],
+			Tag: map[string]string{
+				"amd64": strings.Join(strings.Split(ecOperatorImage, ":")[1:], ":"),
+				"arm64": strings.Join(strings.Split(ecOperatorImage, ":")[1:], ":"),
+			},
+		},
+		"utils": {
+			Repo: strings.Split(ecUtilsImage, ":")[0],
+			Tag: map[string]string{
+				"amd64": strings.Join(strings.Split(ecUtilsImage, ":")[1:], ":"),
+				"arm64": strings.Join(strings.Split(ecUtilsImage, ":")[1:], ":"),
+			},
+		},
+	}
```

Comment:
could you split this out into its own function operatorImages?

Reply:
yep yep, that would make sense